src/test.cpp

```diff
@@ -1407,6 +1410,118 @@ TEST_CASE("AggregationInfo") {
+        // master public key.)
+
+        for (int target = 1; target <= N; ++target) {
+            for (int source = 1; source <= N; ++source) {
+                REQUIRE(Threshold::VerifySecretFragment(
+                    target, frags[source-1][target-1], commits[source-1], T));
+            }
+        }
+
+        // 3. Each player computes the shared, master public key:
+        // masterPubkey = PublicKey::AggregateInsecure(...)
+        // They also create their secret share from all secret
+        // fragments received (now verified):
+        // secretShare = PrivateKey::AggregateInsecure(...)
+
+        PublicKey masterPubkey = PublicKey::AggregateInsecure({
+            commits[0][0], commits[1][0], commits[2][0]
+        });
+
+        // recvdFrags[j][i] = frags[i][j]
+        std::vector<std::vector<PrivateKey>> recvdFrags = {{}};
+        for (int i = 0; i < N; ++i) {
+            recvdFrags.emplace_back(std::vector<PrivateKey>());
+            for (int j = 0; j < N; ++j) {
+                recvdFrags[i].emplace_back(frags[j][i]);
+            }
+        }
+
+        PrivateKey secretShare1 = PrivateKey::AggregateInsecure(recvdFrags[0]);
+        PrivateKey secretShare2 = PrivateKey::AggregateInsecure(recvdFrags[1]);
+        PrivateKey secretShare3 = PrivateKey::AggregateInsecure(recvdFrags[2]);
+
+        // 4a. Player P creates a pre-multiplied signature share wrt T players:
+        // sigShare = Threshold::SignWithCoefficient(...)
+        // These signature shares can be combined to sign the msg:
+        // signature = InsecureSignature::Aggregate(...)
+        // The advantage of this approach is that forming the final signature
+        // no longer requires information about the players.
+
+        uint8_t msg[] = {100, 2, 254, 88, 90, 45, 23};
+        uint8_t hash[32];
+        Util::Hash256(hash, msg, sizeof(msg));
+
+        size_t players[] = {1, 3};
+        // For example, players 1 and 3 sign.
+        InsecureSignature sigShareC1 = Threshold::SignWithCoefficient(
+            secretShare1, msg, (size_t) sizeof(msg), (size_t) 1, players, T);
+        InsecureSignature sigShareC3 = Threshold::SignWithCoefficient(
+            secretShare3, msg, (size_t) sizeof(msg), (size_t) 3, players, T);
+
+        InsecureSignature signature = InsecureSignature::Aggregate({
+            sigShareC1, sigShareC3});
+
+        REQUIRE(signature.Verify({hash}, {masterPubkey}));
+
+        // 4b. Alternatively, players may sign the message blindly, creating
+        // a unit signature share: sigShare = secretShare.SignInsecure(...)
+        // These signatures may be combined with lagrange coefficients to
+        // sign the message: signature = Threshold::AggregateUnitSigs(...)
+        // The advantage to this approach is that each player does not need
+        // to know the final list of signatories.
+
+        // For example, players 1 and 3 sign.
+        InsecureSignature sigShareU1 = secretShare1.SignInsecure(
```

Comment:
I think it's kind of strange that we're using insecure signature, but it's ok because of the verify step. Maybe we should mention this somewhere?

Reply:
done